addons/mail/static/src/core/common/store_service.js

```diff
@@ -420,54 +420,36 @@ export class Store extends BaseStore {
         );
     }
 
-    getMentionsFromText(body, { mentionedChannels = [], mentionedPartners = [] } = {}) {
+    getMentionsFromText(body, mentions) {
         if (this.self.type !== "partner") {
             // mentions are not supported for guests
-            return {};
+            return [];
         }
-        const validMentions = {};
-        const partners = [];
-        const threads = [];
-        for (const partner of mentionedPartners) {
-            const index = body.indexOf(`@${partner.name}`);
-            if (index === -1) {
-                continue;
-            }
-            partners.push(partner);
-        }
-        for (const thread of mentionedChannels) {
-            const index = body.indexOf(`#${thread.displayName}`);
-            if (index === -1) {
-                continue;
+
+        const validMentions = mentions.filter((mention) => {
+            switch (mention.type) {
+                case "channel":
+                    return body.includes(`#${mention.channel.displayName}`);
```

Comment:
`includes` might not be supported by some old browsers. but I like includes more here. need some ideas from seb and aku

Reply:
`includes` is not supported by some older browsers but I believe it has better general coverage than `indexOf`:
https://caniuse.com/?search=indexof
https://caniuse.com/?search=includes

Reply:
@Atovange we support "ever-green" browsers, that is Safari / Chrome / Firefox / Edge. If all of them support a feature, then it's totally ok to use the feature. We do not care for IE or Opera mini at all.